src/python/gem5/resources/resource.py

```diff
@@ -91,12 +93,16 @@ def __init__(
                 f"Local path specified for resource, '{local_path}', does not "
                 "exist."
             )
-
+        self._id = id
+        self._category = category
         self._local_path = local_path
         self._description = description
         self._source = source
         self._version = resource_version
 
+    def __str__(self):
+        return f"{self._category} resource({self._id}, {self._version})\nFor more information, please visit https://resources.gem5.org/resources/{self._id}?version={self._version}"
```

Comment:
I don't think black cares about strings breaking line length rules so i typically do it manually.

```suggestion
        return f"{self._category} resource({self._id}, {self._version})\n"
                "For more information, please visit "
               f"https://resources.gem5.org/resources/{self._id}?"
               f"version{self._version}"
```

In python

```py
a_string = "hello this is an example of a string that may be a bit too long"
```

is the same as

```py
a_string = "hello "
           "this is an"
           " example of a string that may be "
           "a bi"
           "t too "
           "long"
```